libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1009,6 +1009,14 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 		}
 	}
 
+	if (valueTypes.size() != variables.size())
```

Comment:
Does it make sense to do the below in any case?

Reply:
The only problematic case is when we have more variables than values which is what we basically do here.
The other cases is already being taken care of.

I mean.. suppose it wouldn't hurt to do it unconditionally..

Reply:
I am not sure really. If this condition above is true we have already triggered an error.
We're only making it fatal here if the result of that is that a variable is left without a type.

Doing it unconditionally.. means we don't know for sure anymore if an error was already triggered for this problem so should we do BOOST_THROw or a new fatalError? 
I feel this introduces too many complications.